src/ManageContainer/Client/ManageToDbGate/Client.go

```diff
@@ -141,10 +146,20 @@ func (c Client) InsertShares(dataID string, schema []string, pieceID int32, shar
 	return c.insertShares(conn, dataID, schema, pieceID, shares, sent_at)
 }
 
-func (c Client) existPieceID(conn *grpc.ClientConn, dataID string, pieceID int32) (bool, error) {
-	ls.Lock(dataID)
-	defer ls.Unlock(dataID)
-	query := fmt.Sprintf("SELECT x.meta, meta().id FROM `share` x WHERE x.data_id = '%s';", dataID)
+func (c Client) existPieceID(conn *grpc.ClientConn, bucket string, ID string, pieceID int32) (bool, error) {
+	var idName string
+	if bucket == "share" {
+		idName = "data_id"
+	} else if bucket == "result" {
+		idName = "job_uuid"
+	} else {
+		err := fmt.Errorf("'bucket' must be `share` or `result`. %s is not found.", bucket)
+		return false, err
+	}
+
+	ls.Lock(ID)
+	defer ls.Unlock(ID)
+	query := fmt.Sprintf("SELECT x.meta, meta().id FROM `%s` x WHERE x.%s= '%s';", bucket, idName, ID)
```

Comment:
@mdonaka 
There's not much point in doing Escape this time, but you should use it to make sure you don't forget Escape when you really need it.

```suggestion
	query := fmt.Sprintf("SELECT x.meta, meta().id FROM `%s` x WHERE x.%s= '%s';", 
		bucket, utils.EscapeInjection(idName, utils.Where), utils.EscapeInjection(ID, utils.Where))
```

Reply:
Thanks for pointing that out!